person.rb

```diff
@@ -0,0 +1,22 @@
+class Person
+  # getters and setters
+  attr_accessor :name, :age
+  attr_reader :id
+
+  # constructor
+  def initialize(id, age, name = 'unknown', parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```

Comment:
- [ ]  I will suggest you to generate the id instead of the passing it as a argument 👍 

```ruby
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
```

You can generate it this way  , but also you can use the UUID package that can do the same thing 👍